Fix: class clashes

java.lang.NoSuchMethodError: 'apache.rocketmq.v2.MessagingServiceGrpc$MessagingServiceFutureStub apache.rocketmq.v2.MessagingServiceGrpc.newFutureStub(org.apache.rocketmq.shaded.io.grpc.Channel)'
	at org.apache.rocketmq.client.java.rpc.RpcClientImpl.<init>(RpcClientImpl.java:95)
	at org.apache.rocketmq.client.java.impl.ClientManagerImpl.getRpcClient(ClientManagerImpl.java:184)
	at org.apache.rocketmq.client.java.impl.ClientManagerImpl.queryRoute(ClientManagerImpl.java:202)
	at org.apache.rocketmq.client.java.impl.ClientImpl.fetchTopicRoute0(ClientImpl.java:610)
	at org.apache.rocketmq.client.java.impl.ClientImpl.fetchTopicRoute(ClientImpl.java:587)
	at org.apache.rocketmq.client.java.impl.ClientImpl.startUp(ClientImpl.java:187)
	at org.apache.rocketmq.client.java.impl.producer.ProducerImpl.startUp(ProducerImpl.java:114)
	at org.apache.rocketmq.shaded.com.google.common.util.concurrent.AbstractIdleService$DelegateService$1.run(AbstractIdleService.java:62)
	at org.apache.rocketmq.shaded.com.google.common.util.concurrent.Callables.lambda$threadRenaming$3(Callables.java:103)
	at java.base/java.lang.Thread.run(Thread.java:1589)
2023-11-27 11:47:35.324 ERROR [92836] [ProducerImpl-0 STARTING] [o.a.r.c.j.i.p.ProducerImpl#startUp:117] - Failed to start the rocketmq producer, try to shutdown it, clientId=Zhanhui-MBP.lan@92836@0@4g76mzqhad
java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError: 'apache.rocketmq.v2.MessagingServiceGrpc$MessagingServiceFutureStub apache.rocketmq.v2.MessagingServiceGrpc.newFutureStub(org.apache.rocketmq.shaded.io.grpc.Channel)'
	at org.apache.rocketmq.shaded.com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:588)
	at org.apache.rocketmq.shaded.com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:547)
	at org.apache.rocketmq.shaded.com.google.common.util.concurrent.FluentFuture$TrustedFuture.get(FluentFuture.java:91)
	at org.apache.rocketmq.client.java.impl.ClientImpl.startUp(ClientImpl.java:188)
	at org.apache.rocketmq.client.java.impl.producer.ProducerImpl.startUp(ProducerImpl.java:114)
	at org.apache.rocketmq.shaded.com.google.common.util.concurrent.AbstractIdleService$DelegateService$1.run(AbstractIdleService.java:62)
	at org.apache.rocketmq.shaded.com.google.common.util.concurrent.Callables.lambda$threadRenaming$3(Callables.java:103)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.lang.NoSuchMethodError: 'apache.rocketmq.v2.MessagingServiceGrpc$MessagingServiceFutureStub apache.rocketmq.v2.MessagingServiceGrpc.newFutureStub(org.apache.rocketmq.shaded.io.grpc.Channel)'
	at org.apache.rocketmq.client.java.rpc.RpcClientImpl.<init>(RpcClientImpl.java:95)
	at org.apache.rocketmq.client.java.impl.ClientManagerImpl.getRpcClient(ClientManagerImpl.java:184)
	at org.apache.rocketmq.client.java.impl.ClientManagerImpl.queryRoute(ClientManagerImpl.java:202)
	at org.apache.rocketmq.client.java.impl.ClientImpl.fetchTopicRoute0(ClientImpl.java:610)
	at org.apache.rocketmq.client.java.impl.ClientImpl.fetchTopicRoute(ClientImpl.java:587)
	at org.apache.rocketmq.client.java.impl.ClientImpl.startUp(ClientImpl.java:187)
	... 4 common frames omitted

--- a/cli/pom.xml
+++ b/cli/pom.xml
@@ -51,8 +51,26 @@
         </dependency>
         <dependency>
             <groupId>org.apache.rocketmq</groupId>
-            <artifactId>rocketmq-client-java</artifactId>
-            <version>5.0.4</version>
+            <artifactId>rocketmq-client-java-noshade</artifactId>
+            <version>5.0.5</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>io.opentelemetry</groupId>
+                    <artifactId>opentelemetry-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>io.opentelemetry</groupId>
+                    <artifactId>opentelemetry-sdk</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>io.opentelemetry</groupId>
+                    <artifactId>opentelemetry-exporter-otlp-common</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>io.opentelemetry</groupId>
+                    <artifactId>opentelemetry-sdk-metrics</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>io.dropwizard.metrics</groupId>

